### PR TITLE
Add Detail button to show Support Details for each Entitlement

### PR DESCRIPTION
**Pull request by Ali-D-Akbar**
[PROD-2277](https://openedx.atlassian.net/browse/PROD-2277)

A button `Detail` has been added in each row of Entitlement:
Screenshot 2021-03-31 at 7 36 41 PM

Upon clicking the Detail Button, the following modal will be shown containing all the support details for the respective entitlement:
image
Note: The modal is slidable so this screenshot may not be able to show all the fields.

### CODE REVIEW

File: src/users/entitlements/Entitlements.test.jsx [hunk 1]
```diff
@@ -98,7 +121,7 @@ describe('Entitlements Listing', () => {
         const tableData = wrapper.find('table.table');
 
         tableData.find('tbody tr').forEach(row => {
-          const reissueButton = row.find('button.btn-outline-primary').last();
+          const reissueButton = row.find('button#reissue').last();
```
Comment (Ali-D-Akbar) [inline src/users/entitlements/Entitlements.test.jsx:125]: Changed this as there are two buttons with the className `btn-outline-primary` and may cause some confusion.

File: src/users/entitlements/Entitlements.jsx [hunk 2]
```diff
@@ -45,6 +48,20 @@ export default function Entitlements({
...
   }
 
+  // Modal to display Support Details for each Entitlement
+  const openEntitlementModal = (title, supportDetails) => {
+    const tableData = supportDetails.map(supportDetail => ({
+      action: supportDetail.action,
+      comments: supportDetail.comments,
+      actionCreated: formatDate(supportDetail.created),
+      supportUser: supportDetail.supportUser,
+      unenrolledRun: supportDetail.unenrolledRun,
```
Comment (DawoudSheraz) [inline src/users/entitlements/Entitlements.jsx:58]: Sort the data by actionCreated in descending order

File: src/users/entitlements/Entitlements.jsx [hunk 3]
```diff
@@ -193,6 +220,24 @@ export default function Entitlements({
     },
   ];
 
+  const supportDetailsColumn = [
```
Comment (DawoudSheraz) [inline src/users/entitlements/Entitlements.jsx:223]: no need to add sort on the modal table. This data will be limited.

**Review by DawoudSheraz — COMMENTED** [review]
added a few comments to address

File: src/users/entitlements/Entitlements.jsx (hunk 2 above)
Comment (Ali-D-Akbar) [reply src/users/entitlements/Entitlements.jsx:58]: I just checked, the data is always comes in descending order. We don't need to sort it manually.

File: src/users/entitlements/Entitlements.jsx (hunk 3 above)
Comment (Ali-D-Akbar) [reply src/users/entitlements/Entitlements.jsx:223]: removed sorting from the modal table.

File: src/users/entitlements/Entitlements.jsx [hunk 4]
```diff
@@ -45,6 +48,20 @@ export default function Entitlements({
     setCourseSummaryErrors(false);
   }
 
+  // Modal to display Support Details for each Entitlement
+  const openEntitlementModal = (title, supportDetails) => {
```
Comment (DawoudSheraz) [inline src/users/entitlements/Entitlements.jsx]: nit: openEntitlementsSupportDetailsModal

File: src/users/entitlements/Entitlements.jsx [hunk 5]
```diff
@@ -139,8 +167,7 @@ export default function Entitlements({
             </Button>
             <Button
               type="button"
-              className="mt-2"
-              variant="outline-danger"
+              variant="outline-danger mt-2"
```
Comment (DawoudSheraz) [inline src/users/entitlements/Entitlements.jsx:170]: why this change? and are you sure is there a mt-2 variant type?

File: src/users/entitlements/Entitlements.jsx [hunk 6]
```diff
@@ -193,6 +220,24 @@ export default function Entitlements({
...
 
+  const supportDetailsColumn = [
+    {
+      label: 'Action', key: 'action', width: 'col-3',
+    },
+    {
+      label: 'Comments', key: 'comments', width: 'col-3',
+    },
+    {
+      label: 'Action Created', key: 'actionCreated', width: 'col-3',
```
Comment (DawoudSheraz) [inline src/users/entitlements/Entitlements.jsx]: nit: Action Creation Timestamp

File: src/users/entitlements/Entitlements.jsx (hunk 5 above)
Comment (Ali-D-Akbar) [reply src/users/entitlements/Entitlements.jsx:170]: I thought of moving all the className attributes to variants. Also, yes, there's an mt-2 variant type.

File: src/users/entitlements/Entitlements.jsx [hunk 7]
```diff
@@ -27,6 +27,9 @@ export default function Entitlements({
   const [courseSummaryUUID, setCourseSummaryUUID] = useState(null);
   const [courseSummaryData, setCourseSummaryData] = useState(null);
   const [courseSummaryErrors, setCourseSummaryErrors] = useState(false);
+  const [entitlementModalIsOpen, setEntitlementModalIsOpen] = useState(false);
```
Comment (DawoudSheraz) [inline src/users/entitlements/Entitlements.jsx]: nit: make it entitlementDetailModal to be consistent across other related attrs.

**Review by DawoudSheraz — APPROVED** [review]

**Review by azanbinzahid — APPROVED** [review]
